lib/adapters/http/read.js

```diff
@@ -41,6 +48,19 @@ class Iterator {
             this.adapter.trigger('error', err);
         })
         .then(() => {
+            if (this.adapter.followLinkHeader && res.headers['link']) {
+                var parsed = parseLinkHeader(res.headers['link']);
+                if (parsed.next) {
+                    this.adapter.logger.debug('following link header', parsed.next.url);
+                    return this.adapter.fetch(parsed.next.url)
+                    .then((res) => {
+                        this.parse(res, from, to);
+                        res.resume();
```

Comment:
The tests do fail and its for a valid reason. When you send back the `Response` object without first hooking up synchronously the `.on` `data`, `error`, etc. handlers that we do in the `new Iterator`-> `getParser` path then the resolving of the `Response` happens asynchronously up to where we do call the `new Iterator` and by then the `on('data'` for the first few bytes has already been called and that data is gone. Try removing the resume/pause and you'll see. I only had to shift around where the Iterator was being created so we could easily do new `fetch` calls without having to recreate the iterator each time but maybe there's a nicer way to achieve the same thing ?